Library/Homebrew/cli/args.rb

```diff
@@ -241,6 +240,51 @@ def spec(default = :stable)
           default
         end
       end
+
+      def resolve_keg(name)
+        require "keg"
+        require "formula"
+        require "missing_formula"
+
+        raise UsageError if name.empty?
+
+        rack = Formulary.to_rack(name.downcase)
+
+        dirs = rack.directory? ? rack.subdirs : []
+
+        raise NoSuchKegError, rack.basename if dirs.empty?
+
+        linked_keg_ref = HOMEBREW_LINKED_KEGS/rack.basename
+        opt_prefix = HOMEBREW_PREFIX/"opt/#{rack.basename}"
+
+        begin
+          if opt_prefix.symlink? && opt_prefix.directory?
+            Keg.new(opt_prefix.resolved_path)
+          elsif linked_keg_ref.symlink? && linked_keg_ref.directory?
+            Keg.new(linked_keg_ref.resolved_path)
+          elsif dirs.length == 1
+            Keg.new(dirs.first)
+          else
+            f = if name.include?("/") || File.exist?(name)
+              Formulary.factory(name)
+            else
+              Formulary.from_rack(rack)
+            end
+
+            unless (prefix = f.installed_prefix).directory?
+              raise MultipleVersionsInstalledError, rack.basename
+            end
+
+            Keg.new(prefix)
+          end
+        rescue FormulaUnavailableError
+          raise <<~EOS
```

Comment:
```suggestion
          raise MultipleVersionsInstalledError, <<~EOS
```

or a better exception?